### PR TITLE
#327 Remove partial throughput graphs

### PR DESCRIPTION
**Pull request by rmacor**
Backport of https://github.com/radargun/radargun/pull/337 from 2x branch

### CODE REVIEW

File: extensions/cache/src/main/resources/benchmark-dist.xml [hunk 1]
```diff
@@ -45,26 +35,28 @@
...
-      <cache:basic-operations-test test-name="warmup" num-operations="100000" num-threads-per-node="5">
-         <cache:key-selector>
-            <cache:concurrent-keys total-entries="5000" />
-         </cache:key-selector>
-      </cache:basic-operations-test>
-      
-      <repeat from="10" to="30" inc="10">
+      <cache:load num-entries="1000"/>
+      <!---->
+      <!--<cache:basic-operations-test test-name="warmup" num-operations="1000" num-threads-per-node="1">-->
```
Comment (rvansa) [inline extensions/cache/src/main/resources/benchmark-dist.xml]: Is this intentional?

**Comment by rmacor** [comment]
Please review carefully, there are a lot of differences compared to https://github.com/radargun/radargun/commit/e07631476bb4f2cf67162648a6087369fa9f96f0

File: extensions/cache/src/main/resources/benchmark-dist.xml (hunk 1 above)
Comment (rmacor) [inline extensions/cache/src/main/resources/benchmark-dist.xml]: no, that was just for testing, I'll remove it from the PR

File: core/src/main/java/org/radargun/config/Configuration.java [hunk 2]
```diff
@@ -16,13 +16,18 @@
...
    public Configuration(String name) {
       if (name == null) throw new NullPointerException("Configuration name cannot be null");
       this.name = name;
    }
 
+   public String getName() {
+      return name;
+   }
+
+
```
Comment (jmarkos) [inline core/src/main/java/org/radargun/config/Configuration.java]: extra line

File: core/src/main/java/org/radargun/stages/test/TestStage.java [hunk 3]
```diff
@@ -68,7 +71,7 @@
    protected Transactional transactional;
 
    private Completion completion;
-   private OperationSelector operationSelector;
+   public OperationSelector operationSelector;
```
Comment (jmarkos) [inline core/src/main/java/org/radargun/stages/test/TestStage.java]: Does this really need to be public?

File: core/src/main/java/org/radargun/stats/BasicStatistics.java [hunk 4]
```diff
@@ -30,14 +33,33 @@ public BasicStatistics(OperationStats prototype) {
...
    }
 
    public Statistics newInstance() {
       return new BasicStatistics(prototype);
    }
 
    @Override
+   public void registerOperationsGroup(String name, Set<Operation> operations) {
+      if (groupOperationsMap.containsKey(name)) {
+         return;
```
Comment (jmarkos) [inline core/src/main/java/org/radargun/stats/BasicStatistics.java]: The original PR was throwing an exception for this case, any reason you changed that?

File: core/src/main/java/org/radargun/stats/BasicStatistics.java [hunk 5]
```diff
@@ -30,14 +33,33 @@ public BasicStatistics(OperationStats prototype) {
...
+      }
+      for (Map.Entry<String, Set<Operation>> entry : groupOperationsMap.entrySet()) {
+         for (Operation operation : operations) {
+            if (entry.getValue().contains(operation)) {
+               return;
```
Comment (jmarkos) [inline core/src/main/java/org/radargun/stats/BasicStatistics.java]: Same here, the original PR threw an exception.

File: core/src/main/java/org/radargun/stats/PeriodicStatistics.java [hunk 6]
```diff
@@ -17,7 +18,7 @@
 
 /**
  * Keeps a series of {@link Statistics} instances and records the requests according to current timestamp.
- *
+ * <p>
```
Comment (jmarkos) [inline core/src/main/java/org/radargun/stats/PeriodicStatistics.java]: ?

File: extensions/cache/src/main/java/org/radargun/stages/cache/test/CacheInvocations.java [hunk 7]
```diff
@@ -81,7 +81,7 @@ public Operation txOperation() {
    }
 
    public static final class PutWithLifespan<K, V> implements Invocation<V> {
-      private static final Operation TX = TemporalOperations.PUT_WITH_LIFESPAN.derive("TX");
+      public static final Operation TX = TemporalOperations.PUT_WITH_LIFESPAN.derive("TX");
```
Comment (jmarkos) [inline extensions/cache/src/main/java/org/radargun/stages/cache/test/CacheInvocations.java]: Is it necessary to have this changed? (+ other occurences in this file)

File: reporters/reporter-default/src/main/java/org/radargun/reporting/commons/TestAggregations.java [hunk 8]
```diff
@@ -97,23 +99,41 @@ private void addIteration(Report.Test test, List<Aggregation> iterations, Report
...
+               }
+            }
+         }
+
          iterations.add(new Aggregation(nodeStats, nodeThreads, totalStats.get(), totalThreads.get(), it));
          for (String operation : totalStats.get().getOperations()) {
             DefaultOutcome defaultOutcome = totalStats.get().getRepresentation(operation, DefaultOutcome.class);
-            if (defaultOutcome == null || defaultOutcome.requests > 0) operations.add(operation);
+            if (defaultOutcome == null || defaultOutcome.requests > 0)
+               operations.add(operation);
```
Comment (jmarkos) [inline reporters/reporter-default/src/main/java/org/radargun/reporting/commons/TestAggregations.java]: If you're going to refactor stuff like this, I'd add braces, just moving it to a new line feels even more error-prone...

**Comment by rmacor** [comment]
@jmarkos thanks for the review, I've addressed your comments.

**Comment by jmarkos** [comment]
@rmacor Great, thanks!

**Pull request merged**